source/slang/slang-emit-spirv.cpp

```diff
@@ -1745,9 +1884,22 @@ struct SPIRVEmitContext
                 });
             }
         case kIROp_Unreachable:
-            return emitInst(parent, inst, SpvOpUnreachable);
+            return emitOpUnreachable(parent, inst);
         case kIROp_conditionalBranch:
             SLANG_UNEXPECTED("Unstructured branching is not supported by SPIRV.");
+        case kIROp_MakeVector:
```

Comment:
For constructors, we also need:
- VectorReshape
- MatrixReshape
- MakeMatrix
- MakeMatrixFromScalar
- MakeArrayFromElement

We also need these type casts:
- IntCast
- FloatCast
- CastFloatToInt
- CastIntToFloat
Currently these casts can all have vector and matrix operands, so we may need to lower that first.

Reply:
I think the casts are currently handled elsewhere in this module (although we will need to lower the matrix ops)